src/capi/io.rs

```diff
@@ -244,6 +244,8 @@ pub extern "C" fn chewing_set_KBType(ctx: *mut ChewingContext, kbtype: c_int) ->
         KB::ThlPinyin => (AnyKeyboardLayout::qwerty(), Box::new(Pinyin::thl())),
         KB::Mps2Pinyin => (AnyKeyboardLayout::qwerty(), Box::new(Pinyin::mps2())),
         KB::Carpalx => (AnyKeyboardLayout::qwerty(), Box::new(Standard::new())),
+        KB::ColemakDhAnsi => (AnyKeyboardLayout::qwerty(), Box::new(Standard::new())),
+        KB::ColemakDhOrth => (AnyKeyboardLayout::qwerty(), Box::new(Standard::new())),
```

Comment:
Note this will map Colemak to QWERTY. This doesn't block merging your PR because we don't really have a test yet and Rust is not release blocking.

If you want to fix it you need to add a new definition like this

https://github.com/chewing/libchewing/blob/7c032c9567ca1707ea668dddb2c416648ac6b124/src/editor/keyboard/qgmlwy.rs

Reply:
let me try to add it, are there anyway to test if my impl is correct or not?

Reply:
btw, I'm not sure if I can add it correctly, because colemak-dh orth keyboard layout can depend on the user,  they only defined 40 keys in it.

![image](https://github.com/chewing/libchewing/assets/28770408/761b2e0a-bd49-44b7-bc27-5d4cb77db78f)
(figure from: https://colemakmods.github.io/mod-dh/keyboards.html)

Reply:
You can add some tests to `test/test-bopomofo.c` `test_KB()`

If you have rust installed locally you can switch to build and test rust with

```
cmake --preset rust
cmake --build out/build/rust -t test
```

Reply:
Could you elaborate more on the `KeyboardLayout`, how will it affect others? 

For example, Colemak-DH Orth didn't define `-` in the layout, what will happen if I replace them with `Unknown`? And, if the user has different layout than the `KEYCODE_INDEX` (e.g. I define my `-` at mod+t, not at a regular place), will that also affect the output?

Reply:
> btw, I'm not sure if I can add it correctly, because colemak-dh orth keyboard layout can depend on the user, they only defined 40 keys in it.

Yeah, the current design assumed the keyboard is a variation of the us-102 keyboard. Ortholinear keyboard should still work if you leave the other keys in the same place as us-102.

Reply:
> Could you elaborate more on the `KeyboardLayout`, how will it affect others?

KeyboardLayout is used to map key codes to bopomofo key symbols. Some bopomofo keyboard like Standard only look at the key index to determine it's ㄅ or ㄆ or ㄇ.

> For example, Colemak-DH Orth didn't define `-` in the layout, what will happen if I replace them with `Unknown`? And, if the user has different layout than the `KEYCODE_INDEX` (e.g. I define my `-` at mod+t, not at a regular place), will that also affect the output?

I guess as long as you keep the `-` in the same place in the `KEYCODE_INDEX` it should be fine. No other code currently uses position dependent keymaps for other symbols.

Yeah, I realize the current design doesn't work well for ortholiner keyboard. I have plans to refactor the keymap module but this isn't the priority for now. If it doesn't work I'm can merge your PR just fine.

Reply:
added in rust & add basic tests